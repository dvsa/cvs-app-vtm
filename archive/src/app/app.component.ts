import { Component, ChangeDetectionStrategy } from '@angular/core';
import { initAll } from 'govuk-frontend';

@Component({
  selector: 'vtm-app',
  templateUrl: 'app.component.html',
  styleUrls: ['app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AppComponent {
  constructor() {
    initAll();
  }
}
