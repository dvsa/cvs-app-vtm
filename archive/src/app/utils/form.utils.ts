import { FormArray, FormControl, FormGroup, ValidatorFn } from '@angular/forms';

export interface FieldState {
  value: string | boolean | number | FormControl[];
  validators?: ValidatorFn[];
}

export interface FormControlsState {
  name: string;
  fieldState: FieldState;
}
export const FORM_UTILS = {
  addControlsToFormGroup
};

function addControlsToFormGroup(formGroup: FormGroup, listOfFields: FormControlsState[]) {
  return listOfFields.map((field) => {
    formGroup.addControl(
      field.name,
      Array.isArray(field.fieldState.value)
        ? new FormArray(field.fieldState.value, field.fieldState.validators)
        : new FormControl(field.fieldState.value, field.fieldState.validators)
    );
  });
}
