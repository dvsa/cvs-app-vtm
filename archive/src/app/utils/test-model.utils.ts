import { TestResultModel } from '@app/models/test-result.model';
import { TestType } from '@app/models/test.type';
import { TestTypeCategory } from '@app/models/test-type-category';

export const TEST_MODEL_UTILS = {
  mockTestRecord,
  mockTestType,
  mockTestTypeCategory
};

function mockTestRecord(args?: Partial<TestResultModel>): TestResultModel {
  const mock: TestResultModel = {
    reasonForCreation: '',
    testResultId: '123',
    vehicleType: 'hgv',
    systemNumber: '123',
    testTypes: [this.mockTestType()],
    testEndTimestamp: '2020-01-16T12:24:38.027Z',
    testStartTimestamp: '2020-01-16T10:24:38.027Z'
  } as TestResultModel;

  return { ...mock, ...args };
}

function mockTestType(args?: Partial<TestType>): TestType {
  const mock: TestType = {
    testTypeId: '1',
    seatbeltInstallationCheckDate: true,
    testTypeName: 'first test',
    testNumber: '123123',
    testTypeEndTimestamp: '2020-01-16T12:24:38.027Z',
    testExpiryDate: '2021-01-16T12:24:38.027Z',
    modType: {
      code: '1',
      description: 'test'
    }
  } as TestType;

  return { ...mock, ...args };
}

function mockTestTypeCategory(args?: Partial<TestTypeCategory>): TestTypeCategory {
  const mock: TestTypeCategory = {
    id: '1',
    name: 'Annual test',
    linkedIds: ['38', '39'],
    testTypeName: 'Annual test',
    forVehicleType: ['psv'],
    forVehicleSize: ['small', 'large'],
    forVehicleConfiguration: ['articulated', 'rigid'],
    forVehicleAxles: null,
    forEuVehicleCategory: null,
    forVehicleClass: null,
    forVehicleSubclass: null,
    forVehicleWheels: null,
    testTypeClassification: 'Annual With Certificate'
  } as TestTypeCategory;

  return { ...mock, ...args };
}
