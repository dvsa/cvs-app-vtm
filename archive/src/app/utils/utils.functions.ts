export function scrollToSection(sections, sectionTitle) {
  window.scrollTo({ top: document.body.scrollHeight, behavior: 'smooth' });
  sections[sections.findIndex((section) => section.panel === sectionTitle)].isOpened = true;
}

// add  your util fn below
