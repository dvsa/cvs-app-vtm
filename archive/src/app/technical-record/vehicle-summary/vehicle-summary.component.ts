import {
  Component,
  ChangeDetectionStrategy,
  Input,
  SimpleChanges,
  OnChanges
} from '@angular/core';

import { TechRecord } from '@app/models/tech-record.model';
import { TechRecordHelperService } from '../tech-record-helper.service';
import { VIEW_STATE } from '@app/app.enums';

@Component({
  selector: 'vtm-vehicle-summary',
  templateUrl: './vehicle-summary.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class VehicleSummaryComponent implements OnChanges {
  @Input() activeRecord: TechRecord;
  @Input() editState: boolean;
  @Input() viewState: VIEW_STATE;

  vehicleClassDescription: string;
  isStandardVehicle: boolean;

  constructor(public techRecHelper: TechRecordHelperService) {}

  ngOnChanges(changes: SimpleChanges): void {
    const { activeRecord } = changes;

    if (activeRecord) {
      this.vehicleClassDescription =
        this.activeRecord.vehicleClass && this.activeRecord.vehicleClass.description
          ? this.activeRecord.vehicleClass.description
          : null;

      this.isStandardVehicle = this.techRecHelper.isStandardVehicle(
        this.activeRecord.vehicleType
      );
    }
  }

  axlesHasParkingBrakeMrk(): boolean {
    return (
      this.activeRecord.axles &&
      this.activeRecord.axles.length &&
      this.activeRecord.axles.some((x) => x.parkingBrakeMrk)
    );
  }
}
