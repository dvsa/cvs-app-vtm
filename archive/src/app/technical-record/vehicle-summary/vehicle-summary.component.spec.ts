import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Component, Input } from '@angular/core';

import { VehicleSummaryComponent } from '@app/technical-record/vehicle-summary/vehicle-summary.component';
import { SharedModule } from '@app/shared/shared.module';
import { TESTING_UTILS } from '@app/utils/';
import { TechRecord } from '@app/models/tech-record.model';
import { VEHICLE_TYPES, VIEW_STATE } from '@app/app.enums';
import { TechRecordHelperService } from '../tech-record-helper.service';

const getTechRecord = (): TechRecord => {
  return {
    approvalType: 'approval',
    regnDate: '2020-06-08',
    manufactureYear: 2003,
    euVehicleCategory: '3',
    departmentalVehicleMarker: true,
    vehicleClass: { code: '2', description: 'motorbikes over 200cc' },
    axles: [TESTING_UTILS.mockAxle()]
  } as TechRecord;
};

describe('VehicleSummaryComponent', () => {
  let fixture: ComponentFixture<TestVehicleSummaryComponent>;
  let component: TestVehicleSummaryComponent;
  let techRecHelper: TechRecordHelperService;
  let techHelper: jasmine.Spy;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [
        VehicleSummaryComponent,
        TestVehicleSummaryComponent,
        TestVehicleSummaryHgvComponent,
        TestVehicleSummaryTrlComponent,
        TestVehicleSummaryPsvComponent,
        TestVehicleSummaryEditComponent
      ],
      providers: [TechRecordHelperService]
    }).compileComponents();

    techRecHelper = TestBed.get(TechRecordHelperService);
    fixture = TestBed.createComponent(TestVehicleSummaryComponent);
    component = fixture.componentInstance;
    component.activeRecord = {
      ...getTechRecord()
    } as TechRecord;

    techHelper = spyOn(techRecHelper, 'isStandardVehicle');
  });

  it('should create view only with HGV populated data', () => {
    techHelper.and.returnValue(true);
    component.activeRecord.vehicleType = VEHICLE_TYPES.HGV;
    fixture.detectChanges();

    expect(techRecHelper.isStandardVehicle).toHaveBeenCalledWith(VEHICLE_TYPES.HGV);
    expect(fixture).toMatchSnapshot();
  });

  it('should create view only with TRL populated data', () => {
    techHelper.and.returnValue(true);
    component.activeRecord = {
      vehicleType: VEHICLE_TYPES.TRL,
      firstUseDate: '2019-05-16'
    } as TechRecord;
    fixture.detectChanges();

    expect(techRecHelper.isStandardVehicle).toHaveBeenCalledWith(VEHICLE_TYPES.TRL);
    expect(fixture).toMatchSnapshot();
  });

  it('should create view only with PSV populated data', () => {
    techHelper.and.returnValue(true);
    component.activeRecord = {
      vehicleType: VEHICLE_TYPES.PSV,
      seatbeltInstallationApprovalDate: '2018-05-16'
    } as TechRecord;
    fixture.detectChanges();

    expect(techRecHelper.isStandardVehicle).toHaveBeenCalledWith(VEHICLE_TYPES.PSV);
    expect(fixture).toMatchSnapshot();
  });

  it('should check if axles has no parking brake mrk', () => {
    const componentVS = TestBed.createComponent(VehicleSummaryComponent).componentInstance;
    componentVS.activeRecord = getTechRecord();

    expect(componentVS.axlesHasParkingBrakeMrk()).toBeTruthy();
  });

  it('should render the editable component if editState is true', () => {
    component.editState = true;
    component.viewState = VIEW_STATE.EDIT;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  selector: 'test-vtm-vehicle-summary',
  template: `
    <vtm-vehicle-summary
      [activeRecord]="activeRecord"
      [editState]="editState"
      [viewState]="viewState"
    >
    </vtm-vehicle-summary>
  `
})
class TestVehicleSummaryComponent {
  activeRecord: TechRecord;
  editState: boolean;
  viewState = VIEW_STATE.VIEW_ONLY;
}

@Component({
  selector: 'vtm-vehicle-summary-hgv',
  template: `
    <div>HGV active record is: {{ activeRecord | json }}</div>
  `
})
class TestVehicleSummaryHgvComponent {
  @Input() activeRecord: TechRecord;
}

@Component({
  selector: 'vtm-vehicle-summary-trl',
  template: `
    <div>TRL active record is: {{ activeRecord | json }}</div>
  `
})
class TestVehicleSummaryTrlComponent {
  @Input() activeRecord: TechRecord;
}

@Component({
  selector: 'vtm-vehicle-summary-psv',
  template: `
    <div>PSV active record is: {{ activeRecord | json }}</div>
  `
})
class TestVehicleSummaryPsvComponent {
  @Input() activeRecord: TechRecord;
}

@Component({
  selector: 'vtm-vehicle-summary-edit',
  template: `
    <div>active record is: {{ techRecord | json }}</div>
    <div>Current view is: {{ viewState }} Edit mode</div>
  `
})
class TestVehicleSummaryEditComponent {
  @Input() techRecord: TechRecord;
  @Input() viewState: VIEW_STATE;
}
