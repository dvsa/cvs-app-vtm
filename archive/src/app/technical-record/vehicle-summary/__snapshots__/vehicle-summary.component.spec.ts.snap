// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleSummaryComponent should create view only with HGV populated data 1`] = `
<test-vtm-vehicle-summary
  activeRecord={[Function Object]}
  viewState="0"
>
  <vtm-vehicle-summary
    ng-reflect-active-record="[object Object]"
    ng-reflect-view-state="0"
  >
    
    
    <dl
      class="govuk-summary-list"
    >
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle type 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleType"
        >
           HGV 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Date of first registration 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-regnDate"
        >
          
          <span>
             08/06/2020 
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Year of manufacture 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-manufactureYear"
        >
           2003 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Number of axles 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-noOfaxles"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           DTP number 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-dtpNumber"
        >
          
          <span>
            -
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Axles fitted with a parking brake 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-parkingBrakeMrk"
        >
          
          
          
          <div>
            
            <div>
              <span
                id="test-parkingBrakeMrkAxle-1"
              >
                Axle 1
              </span>
            </div>
          </div>
          
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle class 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleClassDescription"
        >
           Motorbikes over 200cc 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle configuration 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleConfiguration"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           EU vehicle category 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-euVehicleCategory"
        >
           3 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Departmental vehicle marker 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-departmentalVehicleMarker"
        >
           Yes 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Alteration marker 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-alterationMarker"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
    </dl>
    
    
    <vtm-vehicle-summary-hgv
      ng-reflect-active-record="[object Object]"
    >
      <div>
        HGV active record is: {
    "approvalType": "approval",
    "regnDate": "2020-06-08",
    "manufactureYear": 2003,
    "euVehicleCategory": "3",
    "departmentalVehicleMarker": true,
    "vehicleClass": {
      "code": "2",
      "description": "motorbikes over 200cc"
    },
    "axles": [
      {
        "parkingBrakeMrk": true,
        "axleNumber": 1,
        "brakes": {
          "brakeActuator": 123,
          "leverLength": 123,
          "springBrakeParking": true
        },
        "tyres": {
          "tyreSize": "16",
          "plyRating": "12S",
          "fitmentCode": "12313",
          "dataTrAxles": 1234
        },
        "weights": {
          "gbWeight": 100,
          "kerbWeight": 150,
          "designWeight": 250,
          "eecWeight": 150
        }
      }
    ],
    "vehicleType": "hgv"
  }
      </div>
    </vtm-vehicle-summary-hgv>
    
    
    
    <dl
      class="govuk-summary-list"
    >
      <h2
        class="govuk-heading-m"
      >
        Type Approval
      </h2>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Approval type
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-approvalType"
        >
           approval 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Approval type number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-approvalTypeNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          National type approval number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-ntaNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Variant number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-variantNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Variant version number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-variantVersionNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      
      
    </dl>
    
    
    
  </vtm-vehicle-summary>
</test-vtm-vehicle-summary>
`;

exports[`VehicleSummaryComponent should create view only with PSV populated data 1`] = `
<test-vtm-vehicle-summary
  activeRecord={[Function Object]}
  viewState="0"
>
  <vtm-vehicle-summary
    ng-reflect-active-record="[object Object]"
    ng-reflect-view-state="0"
  >
    
    
    <dl
      class="govuk-summary-list"
    >
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle type 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleType"
        >
           PSV 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Date of first registration 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-regnDate"
        >
          
          <span>
            -
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Year of manufacture 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-manufactureYear"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Number of axles 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-noOfaxles"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           DTP number 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-dtpNumber"
        >
          
          <span>
            -
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Axles fitted with a parking brake 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-parkingBrakeMrk"
        >
          
          <span>
            -
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle class 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleClassDescription"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle configuration 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleConfiguration"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           EU vehicle category 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-euVehicleCategory"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Departmental vehicle marker 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-departmentalVehicleMarker"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Alteration marker 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-alterationMarker"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
    </dl>
    
    
    
    <vtm-vehicle-summary-psv
      ng-reflect-active-record="[object Object]"
    >
      <div>
        PSV active record is: {
    "vehicleType": "psv",
    "seatbeltInstallationApprovalDate": "2018-05-16"
  }
      </div>
    </vtm-vehicle-summary-psv>
    
    
    <dl
      class="govuk-summary-list"
    >
      <h2
        class="govuk-heading-m"
      >
        Type Approval
      </h2>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Approval type
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-approvalType"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Approval type number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-approvalTypeNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          National type approval number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-ntaNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Variant number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-variantNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Variant version number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-variantVersionNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           COIF serial number 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-coifSerialNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           COIF certifier name 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-coifCertifierName"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           COIF date 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-coifDate"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
    </dl>
    
    
    
  </vtm-vehicle-summary>
</test-vtm-vehicle-summary>
`;

exports[`VehicleSummaryComponent should create view only with TRL populated data 1`] = `
<test-vtm-vehicle-summary
  activeRecord={[Function Object]}
  viewState="0"
>
  <vtm-vehicle-summary
    ng-reflect-active-record="[object Object]"
    ng-reflect-view-state="0"
  >
    
    
    <dl
      class="govuk-summary-list"
    >
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle type 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleType"
        >
           TRL 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Date of first registration 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-regnDate"
        >
          
          <span>
            -
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Year of manufacture 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-manufactureYear"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Number of axles 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-noOfaxles"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           DTP number 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-dtpNumber"
        >
          
          <span>
            -
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Axles fitted with a parking brake 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-parkingBrakeMrk"
        >
          
          <span>
            -
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle class 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleClassDescription"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Vehicle configuration 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-vehicleConfiguration"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           EU vehicle category 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-euVehicleCategory"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Departmental vehicle marker 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-departmentalVehicleMarker"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
           Alteration marker 
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-alterationMarker"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
    </dl>
    
    
    
    
    <vtm-vehicle-summary-trl
      ng-reflect-active-record="[object Object]"
    >
      <div>
        TRL active record is: {
    "vehicleType": "trl",
    "firstUseDate": "2019-05-16"
  }
      </div>
    </vtm-vehicle-summary-trl>
    
    <dl
      class="govuk-summary-list"
    >
      <h2
        class="govuk-heading-m"
      >
        Type Approval
      </h2>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Approval type
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-approvalType"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Approval type number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-approvalTypeNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          National type approval number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-ntaNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Variant number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-variantNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Variant version number
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-variantVersionNumber"
        >
           - 
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
      
      
      
    </dl>
    
    
    
  </vtm-vehicle-summary>
</test-vtm-vehicle-summary>
`;

exports[`VehicleSummaryComponent should render the editable component if editState is true 1`] = `
<test-vtm-vehicle-summary
  activeRecord={[Function Object]}
  editState={[Function Boolean]}
  viewState={[Function Number]}
>
  <vtm-vehicle-summary
    ng-reflect-active-record="[object Object]"
    ng-reflect-edit-state="true"
    ng-reflect-view-state="1"
  >
    
    
    
    
    <dl
      class="govuk-summary-list"
    >
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          <vtm-vehicle-summary-edit
            ng-reflect-tech-record="[object Object]"
            ng-reflect-view-state="1"
          >
            <div>
              active record is: {
    "approvalType": "approval",
    "regnDate": "2020-06-08",
    "manufactureYear": 2003,
    "euVehicleCategory": "3",
    "departmentalVehicleMarker": true,
    "vehicleClass": {
      "code": "2",
      "description": "motorbikes over 200cc"
    },
    "axles": [
      {
        "parkingBrakeMrk": true,
        "axleNumber": 1,
        "brakes": {
          "brakeActuator": 123,
          "leverLength": 123,
          "springBrakeParking": true
        },
        "tyres": {
          "tyreSize": "16",
          "plyRating": "12S",
          "fitmentCode": "12313",
          "dataTrAxles": 1234
        },
        "weights": {
          "gbWeight": 100,
          "kerbWeight": 150,
          "designWeight": 250,
          "eecWeight": 150
        }
      }
    ]
  }
            </div>
            <div>
              Current view is: 1 Edit mode
            </div>
          </vtm-vehicle-summary-edit>
        </dt>
      </div>
    </dl>
  </vtm-vehicle-summary>
</test-vtm-vehicle-summary>
`;
